examples/shouldi/shouldi/dependency_check.py

```diff
@@ -0,0 +1,53 @@
+import json
+import asyncio
+from typing import Dict, Any
+
+from dffml.df.base import op
+from dffml.df.types import Definition
+
+package_src_dir = Definition(name="package_src_dir", primitive="str")
+dependency_check_output = Definition(
+    name="dependency_check_output", primitive="Dict[str, Any]"
+)
+
+
+class DependencyCheckError(Exception):
+    """
+    Raised when dependency-check fails
+    """
+
+
+@op(
+    inputs={"pkg": package_src_dir},
+    outputs={"report": dependency_check_output},
+)
+async def run_dependency_check(pkg: str) -> Dict[str, Any]:
+    """
+    CLI usage: dffml service dev run -log debug shouldi.dependency_check:run_dependency_check -pkg .
+    """
+    proc = await asyncio.create_subprocess_exec(
+        "dependency-check",
+        "-f",
+        "JSON",
+        "-s",
+        "./...",
```

Comment:
@pdxjohnny a question: what does `./...` does here, also does `cwd=pkg` prints the whole address of that package?
We want to run command like - 
`dependency-check -f JSON -s "*complete address of the package*" `

Reply:
You don't want `./...` here, that's specific to golang. You just want `.` which says scan the current directory. We then set the current directory by using `cwd=pkg` (where pkg is the directory). We should probably add some logic to this operation to check if `pkg` is a file, and then set `cwd` to the directory the file is in if so, and instead of `.` pass the filename.